### PR TITLE
R: add maintainer(s)

### PR DESCRIPTION
**Pull request by adamjstewart**
I talked to @glennpj and he offered to be a maintainer for Spack's R recipe and R packages.

Would anyone else like to join him? You don't have to be a Spack expert, it just gives us someone to ping when users report build issues or submit PRs related to R or R packages. You can also choose to maintain only a subset of R packages if you don't want to maintain all of them.

Pinging some other known Spack + R users: @HenrikBengtsson @JavierCVilla @codeandkey @odoublewen @hartzell @cosmicexplorer 

Some ideas for future work related to R:

* Create a `cran` attribute similar to #17587 to simplify homepage/url/list_url
* Add package tests similar to #20023 to test installations
* Create bundle packages for Bioconductor so you can install `bioconductor@3.12`
* Automate adding new packages/versions or integrate with the builtin package manager

P.S. If you get a chance, take a look at https://github.com/spack/spack/labels/r and see if you can close any of these issues

### CODE REVIEW

**Comment by adamjstewart** [comment]
Also, let me know if anyone wants to join the equivalent Python PR: #21125

**Review by alalazo — APPROVED** [review]

File: lib/spack/spack/build_systems/r.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ class RPackage(PackageBase):
     """
     phases = ['install']
 
+    maintainers = ['glennpj']
```
Comment (glennpj) [inline lib/spack/spack/build_systems/r.py:28]: If a `maintainers` field is put in any specific `r-*` package then that will remove the maintainer value set here for that package. I have not traced this but it seems that we want to be able to concatenate the `maintainers` list.

Comment (adamjstewart) [reply lib/spack/spack/build_systems/r.py:28]: Correct. Personally, I feel like if someone who actually develops a particular R packages wants to maintain it, they'll probably be better than you or I. You can always ask them to add you to the package-specific maintainers field. I don't see an easy way around this.

**Comment by odoublewen** [comment]
Although I did update and create a number of smallish number of `r-*` packages, I quickly came to the conclusion that it was a truly Sisyphean task.  There's just too many of them, too many interdependencies, and they get updated too often.  Plus (and to me, this is the important part):  R has it's own package management system (warts and all, though it's recently been [getting better](https://rstudio.github.io/renv/articles/renv.html).)  So, I stopped updating and using most of the `r-*` packages because it felt like futile busy-work and also like re-inventing a fairly good wheel.

I guess an exception to this would be `r-*` packages that support a command-line tool, like [Trinity](https://spack.readthedocs.io/en/latest/package_list.html#trinity) (bioinformatics has a lot of ball-of-mud projects like this that are nonetheless important.)  In such cases, it's nice to be able to get all the sub-package dependencies as a kind of vendored install.

My 2 cents, and I realize it may not even be worth that much.  I have the same opinion about maintaining python packages in Spack.

So... I guess before I add myself as a maintainer for any specific R or python package, I would want to understand:  Is it Spack's intent to try to capture and maintain R packages "for their own sake" or just when they are in service of some other "application package"?

**Comment by adamjstewart** [comment]
There certainly are other package managers for R, just like there are for Python. I don't know as much about R, but for Python, we have:

* pip: can't build non-Python dependencies
* conda: concretizer hell

For this reason, many people including myself use Spack instead of the alternatives. I'm not sure what the landscape is like with R, but we also allow people to install both `py-pip` and `miniconda`/`anaconda` with Spack if they want to. Personally, my recommendation for Python is to use Spack for all packages that require non-Python deps and use pip for everything else.

I believe @alalazo had some thoughts on how we could further integrate language-specific package managers with Spack.

**Comment by glennpj** [comment]
External dependencies can not be installed with R package managers. 

> Personally, my recommendation for Python is to use Spack for all packages that require non-Python deps and use pip for everything else.

The problem with R is that once you build the r-package dependencies of the R packages with non-R deps you wind up building quite a few r-packages.

> I believe @alalazo had some thoughts on how we could further integrate language-specific package managers with Spack.

Integration would be awesome.

**Comment by odoublewen** [comment]
> External dependencies can not be installed with R package managers.

True.  My approach to this has been trial-and-error:
1. `spack install r`
2. In R, `install.packages('foo')`
3. If error, figure out what deps are lacking, and do `spack install foo_dep`
4. go back to 2.

I've come up with a list of Spack packages that make it possible to build and install everything I need in R.   (in case you're interested:  `spack install r openmpi openblas cairo freeglut gtkplus xfs libxt libxml2 libgit2 curl openssl glpk fftw jags mariadb mesa lzma lz4 lzo`)

**Comment by glennpj** [comment]
I have deployed R stacks both ways,

1. As a monolithic stack, using the Bioconductor installer and managing the external dependencies, either with or without spack to manage the external packages.
2. Individual Spack r-packages, everything built with Spack.

I do like the concept of having the individual packages but there is definitely pain involved.

**Pull request merged**